src/react/components/pages/signIn/signInPage.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { cloneElement } from "react"
+import { ISignIn, IAuth } from "../../../../models/applicationState";
+import SignInForm from "./signInForm";
+import { Route, Redirect } from "react-router-dom";
+import ApiService, { ILoginRequestPayload } from "../../../../services/apiService"
+import IAuthActions, * as authActions from "../../../../redux/actions/authActions";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { IApplicationState } from "../../../../models/applicationState";
+import history from "../../../../history"
+import { toast } from "react-toastify";
+
+export interface ISignInPageProps extends React.Props<SignInPage> {
+    actions: IAuthActions;
+    signin: ISignIn;
+}
+
+export interface ISignInPageState {
+    signin: ISignIn;
+    loginRequestPayload: ILoginRequestPayload;
+    auth: IAuth;
+}
+
+function mapStateToProps(state: IApplicationState) {
+    return {
+        auth: state.auth,
+    };
+}
+
+function mapDispatchToProps(dispatch) {
+    return {
+        actions: bindActionCreators(authActions, dispatch),
+    };
+}
+
+@connect(mapStateToProps, mapDispatchToProps)
+export default class SignInPage extends React.Component<ISignInPageProps, ISignInPageState> {
+    constructor(props){
+        super(props);
+        this.state = { 
+            signin: null,
+            loginRequestPayload: null,
+            auth: null,
+        };
+        ApiService.removeToken();
+        this.onFormSubmit = this.onFormSubmit.bind(this);
+
+    }
+
+    private onFormSubmit(signin: ISignIn) {
+        this.setState({
+            loginRequestPayload: {
+                username: signin.email,
+                password: signin.password,
+            }
+        }, () => {
+            this.sendCredentials(signin.remember)
+        })
+    }
+
+    private async sendCredentials(remember: boolean) {
+        try {
+            const token = await ApiService.loginWithCredentials(this.state.loginRequestPayload);
+            localStorage.setItem("token", token.data.access_token);
+            const userInfo = await ApiService.getCurrentUser();
```

Comment:
If you want to make it more fancy, you could do it that way:

I would consider having this Api call as a side-effect of successful `loginWithCredentials` - maybe in Redux Thunk? 

So you could do it that way:
1. You call redux action `LoginWithCredentialsRequest`
2. Redux Thunk function captures that action and executes `ApiService.loginWithCredentials` with await.
3. If failure, dispatch `LoginWithCredentialsFailure` with error message. Create a reducer, which captures that error and put it to `state.auth.error`
3.1. Show `state.auth.error` somewhere in the component
4. If success, dispatch `LoginWithCredentialsSuccess`
5. Dispatch also `ApiService.getCurrentUser()` in the Thunk if login succeeded